File: Example/Example-iOS/ViewController.m
//
//  ViewController.m
//  Example-iOS
//
//  Created by Takeru Chuganji on 1/4/15.
//
//

#import "ViewController.h"
#import "HCImage+BPG.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIImageView *imageView;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    NSString *filePath = [[NSBundle mainBundle] pathForResource:@"sample" ofType:@"bpg"];
    NSData *data = [NSData dataWithContentsOfFile:filePath];
    self.imageView.image = [UIImage imageWithBPGData:data];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Example/iOS/Example/ViewController.swift
//
//  ViewController.swift
//  Example
//
//  Created by Takeru Chuganji on 11/11/15.
//  Copyright © 2015 Takeru Chuganji. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet private weak var imageView: UIImageView?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.imageView?.image = NSBundle.mainBundle().pathForResource("sample", ofType: "bpg")
            .flatMap { NSData(contentsOfFile: $0) }
            .flatMap { UIImage(BPGData: $0) }
    }
}

